src/App.tsx

```diff
@@ -1,93 +1,93 @@
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    const save = (v: T): void => {
+      setValue(v);
+      localStorage.setItem(key, JSON.stringify(v));
+    };
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    return [value, save];
+  }
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-editing" />
-              <label htmlFor="toggle-editing">zxcvbnm</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const [userEmail, setUserEmail] = useState('');
+  const [userName, setUserName] = useState('');
+  const [errorInput, setErrorInput] = useState<ErrorInput>(ErrorInput.NONE);
+  const [secVisIn, setSecVisIn] = useState(false);
+  const [user, setUser] = useLocaleStorage<User[] | []>('user', []);
 
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view2" />
-              <label htmlFor="toggle-view2">1234567890</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+  const addNewUser = (name:string, email:string) => {
+    return client.post('/users', { name, email });
+  };
 
-      <footer className="footer">
-        <span className="todo-count" data-cy="todosCounter">
-          3 items left
-        </span>
+  const checkedUserEmail = () => {
+    client.get<User[]>(`/users?email=${userEmail}`).then(u => setUser(u));
+  };
 
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
+  const formSubmit = (e:React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
 
-          <li>
-            <a href="#/active">Active</a>
-          </li>
+    const p = new RegExp('^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+.[a-zA-Z]{2,}$')
+      .test(userEmail);
 
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
+    if (p) {
+      checkedUserEmail();
+      setSecVisIn(true);
+      setErrorInput(ErrorInput.NONE);
+      if (userName.length >= 4) {
```

Comment:
both inputs can be invalid at the moment
instead of nested `else if` check both - if anyone is invalid set the correct value and return (to prevent running rest code in the function)